bson/doc.go

```diff
@@ -19,24 +20,96 @@
 // 		i32, ok := val.Int32OK()
 // 		// do something with i32...
 //
-// The D family of types is used to build concise representations of BSON using native Go types.
-// These types do not support automatic lookup.
+// Native Go Types
```

Comment:
What makes D and M native, in comparison to bson.Raw?  Do you mean that they are aliases for Go slice and map, respectively?

Reply:
Yeah, the idea was that they're Go representations of BSON, rather than raw BSON bytes.

Reply:
Ack